symphony-bdk-template/symphony-bdk-template-api/src/main/java/com/symphony/bdk/core/template/api/Template.java

```diff
@@ -0,0 +1,18 @@
+package com.symphony.bdk.core.template.api;
+
+import java.util.Map;
+
+/**
+ * Interface to represent a template.
+ * A template takes parameters in input and outputs a string.
+ */
+public interface Template {
+
+  /**
+   *
+   * @param parameters the map which contains the parameter names as map key and the parameter values as map values
+   * @return the generated string where each parameter is replaced by its value provided in the map
+   * @throws TemplateException in case of issues during the string generation, e.g. missing parameter
+   */
+  String process(Map<String, String> parameters) throws TemplateException;
```

Comment:
With a `Map<String, String>` we can only handle 1 fields level in templates. What happen if a template engine accepts complex objects (`${user.address.street}` for example)?

Reply:
👍